### ISSUE TITLE
Issue with methods executed when iterating through a List

### ISSUE THREAD

**Issue by blmj** [nus-cs2030/2122-s2#42]
For Level 6, assuming loaders is a list of Loaders and RecycledLoaders, I used the below for loop:

for (Loader loader : loaders) 

However, when I reach a loader that happens to be a RecycledLoader, the methods executed are taken from the Loader class instead of the RecycledLoader class. Is anybody able to explain why this happens as I thought it was covered during Lecture that writing "Loader" before it only checks whether the methods exist when compiling, and that the actual method run will be from the RecycledLoader class? Thank you!

**Comment by yangyee-hub** [comment]
Did you check that 1) You establish a parent child rs between Loader and Recycled Loader and 2) You override the method getNextAvailableTime() in the loader class properly?

**Comment by shotnothing** [comment]
Hi, there should be no issue as the methods will be overridden. I have tested
`for (Loader loader : loaders)`
on my code and it works fine. 

Do additionally check the recycled loaders work fine standalone and you did not call this.method() in the parent class expecting that it would call the overridden method() in the child class.

Also, given that loaders are ID'ed sequentially, wouldn't a 
`for (int i = 0; i < numLoaders; i++) `
where `ID = i` be easier for you?

**Comment by haoyangw** [comment]
> Did you check that 1) You establish a parent child rs between Loader and Recycled Loader and 2) You override the method getNextAvailableTime() in the loader class properly?

Yeah check that your overriding method in your RecycledLoader has the same: 1) method name, 2) number of arguments, 3) type of arguments, 4) order of arguments. 

To be extra sure, add the '@Override' tag to all your overriding methods in RecycledLoader(if you haven't yet) so there'd be a compile-time error if your RecyledLoader doesn't override Loader(meaning if your code compiles fine then RecycledLoader is correctly overriding, and you can confirm that your issue is elsewhere)

**Comment by shotnothing** [comment]
> To be extra sure, add the '@OverRide' tag to all your overriding methods in RecycledLoader(if you haven't yet) so there'd be a compile-time error if your RecyledLoader doesn't override Loader(meaning if your code compiles fine then RecycledLoader is correctly overriding, and you can confirm that your issue is elsewhere)

Hi I think it should be `@Override`, not sure if the case matters but  override is a standalone word and this case seems to be the standard for annotations.
https://docs.oracle.com/javase/tutorial/java/annotations/predefined.html

**Comment by undeser** [comment]
Hi, think you might have missed the `@Override` tag in your RecycledLoader class causing it to execute methods from the parent class instead.

**Comment by weekiatism** [comment]
Ensure you have declared a parent child relationship by 'extends loaders' 
and indicated the @override tag before your method
